File: 104. Maximum Depth of Binary Tree/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function(root) {
  function dfs(current){
    if (current == undefined){
      return 0;
    }
    let leftDepth = dfs(current.left);
    let rightDepth = dfs(current.right);
    let maxDepth = leftDepth;
    if (rightDepth > maxDepth){
      maxDepth = rightDepth;
    }
    return maxDepth + 1;
  }

  return dfs(root);
};

File: 104. Maximum Depth of Binary Tree/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
  int dfs(TreeNode* current){
    if (current == nullptr){
      return 0;
    }
    int leftDepth = dfs(current->left);
    int rightDepth = dfs(current->right);
    int maxDepth = leftDepth;
    if (rightDepth > maxDepth){
      maxDepth = rightDepth;
    }
    return maxDepth + 1;
  }
public:
  int maxDepth(TreeNode* root) {
    return dfs(root);
  }
};